packages/core/src/compose-tunnel-agent-client.ts

```diff
@@ -92,16 +97,24 @@ export const addComposeTunnelAgentService = (
   },
 })
 
-export const queryTunnels = async ({ retryOpts = { retries: 0 }, tunnelUrlsForService }: {
+export const queryTunnels = async ({
+  retryOpts = { retries: 0 },
+  tunnelUrlsForService,
+  credentials,
+  includeAccessCredentials,
```

Comment:
> Basically, the credentials and includeAccessCredentials are external to queryTunnels to remove dependencies and give more flexibility to the calling client. (in this case, the calling can't separate the credentials from the urls)

credentials are now needed to query the tunnels (because the compose tunnel agent is private)

extracting `includeAccessCredentials` is still possible but not worth it IMO.